salt/states/locale.py

```diff
@@ -37,21 +42,26 @@ def system(name):
            'changes': {},
            'result': None,
            'comment': ''}
-    if __salt__['locale.get_locale']() == name:
-        ret['result'] = True
-        ret['comment'] = 'System locale {0} already set'.format(name)
-        return ret
-    if __opts__['test']:
-        ret['comment'] = 'System locale {0} needs to be set'.format(name)
-        return ret
-    if __salt__['locale.set_locale'](name):
-        ret['changes'] = {'locale': name}
-        ret['result'] = True
-        ret['comment'] = 'Set system locale {0}'.format(name)
-        return ret
-    else:
+    try:
+        if __salt__['locale.get_locale']() == name:
+            ret['result'] = True
+            ret['comment'] = 'System locale {0} already set'.format(name)
+            return ret
+        if __opts__['test']:
+            ret['comment'] = 'System locale {0} needs to be set'.format(name)
+            return ret
+        if __salt__['locale.set_locale'](name):
+            ret['changes'] = {'locale': name}
+            ret['result'] = True
+            ret['comment'] = 'Set system locale {0}'.format(name)
+            return ret
+        else:
+            ret['result'] = False
+            ret['comment'] = 'Failed to set system locale to {0}'.format(name)
+            return ret
+    except CommandNotFoundError as err:
```

Comment:
@sjorge, sorry to bug you about this again, but in the module you're using `CommandExecutionError` whereas in the state it is `CommandNotFoundError`.